packages/bsky/src/hydration/hydrator.ts

```diff
@@ -737,6 +752,33 @@ export class Hydrator {
     return { follows, followBlocks }
   }
 
+  async hydrationBidirectionalBlocks(
+    didMap: Map<string, string[]>,
+  ): Promise<BidirectionalBlocks> {
+    const pairs: RelationshipPair[] = []
+
+    for (const [uri, followers] of didMap) {
+      for (const follower of followers) {
+        pairs.push([didFromUri(uri), follower])
+      }
+    }
+
+    const blocks = await this.graph.getBidirectionalBlocks(pairs)
+    const result = new HydrationMap<HydrationMap<boolean>>()
+
+    for (const [did] of didMap) {
+      const followBlocks = new HydrationMap<boolean>()
+
+      for (const followerDid of didMap.get(did)!) {
```

Comment:
I think there's a nice way to not need the non-null assertion and lean on the types:
```suggestion
    for (const [did, targetDids] of didMap) {
      const followBlocks = new HydrationMap<boolean>()

      for (const followerDid of targetDids) {
```

Reply:
Oy 🤦